defaults.go

```diff
@@ -157,6 +157,20 @@ func setField(field reflect.Value, defaultVal string) error {
 		}
 		callSetter(ref.Interface())
 		field.Set(ref.Elem())
+	case reflect.Slice:
+		for j := 0; j < field.Len(); j++ {
+			sliceItem := field.Index(j)
+			if sliceItem.Kind() != reflect.Struct {
```

Comment:
Might want to call `setField` recursively?

Reply:
Yes, definitely. Updated commit.